src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -261,6 +258,26 @@ private unsafe bool HandleDocuments(ActionType actionType, DocumentsOperationCon
             return moreWorkFound;
         }
 
+        private IEnumerable<Document> GetDocumentFromCollectionThatReference(DocumentsOperationContext databaseContext, TransactionOperationContext indexContext, string collection, Reference referencedDocument, long lastIndexedEtag)
```

Comment:
Can we avoid this? This will conflict in 5.0 and I do not see any benefit at the moment.

Reply:
This is useful when we have a document that is referenced by many documents

Reply:
Ok, I haven't noticed that you have removed the List.